v2/batch.go

```diff
@@ -0,0 +1,116 @@
+package jsonrpc
+
+import (
+	"bytes"
+	"errors"
+	"net/http"
+	"strconv"
+	"sync"
+)
+
+type batchCall struct {
+	call   *clientCall
+	result interface{}
+	id     string
+}
+
+// Batch represents a collection of method calls that will be sent to the
+// server in a single HTTP call.
+type Batch struct {
+	order         []*batchCall
+	calls         map[string]*batchCall
+	id            int
+	mtx           *sync.Mutex
+	DiscardErrors bool
+}
+
+func (batch *Batch) nextID() string {
```

Comment:
I'm guessing the only instance when you want to convert id to string is when you're serializing batch when generating response, internally it would make more sense to operate on int